src-cpp-structure/projects/App/Modules/Module.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "ModuleInterface.h"
+#include "Startup/ModuleBuilder.h"
+
+#include <memory>
+
+namespace Status::Modules
+{
+
+    class RootModule final : public ModuleAccessInterface
+            , public Startup::ModuleDelegateInterface
+            , public std::enable_shared_from_this<RootModule>
+    {
+    public:
+        RootModule(Startup::ModuleBuilder moduleBuilder);
```

Comment:
When we derive from `std::enable_shared_from_this` it's safer to disallow creating objects of this class through constructor.

```cpp
struct Best : std::enable_shared_from_this<Best> // note: public inheritance
{
    [std::shared_ptr]<Best> getptr() {
        return shared_from_this();
    }
    // No public constructor, only a factory function,
    // so there's no way to have getptr return nullptr.
    [[nodiscard]] static [std::shared_ptr]<Best> create() {
        // Not using std::make_shared<Best> because the c'tor is private.
        return [std::shared_ptr]<Best>(new Best());
    }
private:
    Best() = default;
};
```
source: https://en.cppreference.com/w/cpp/memory/enable_shared_from_this